source/common/router/debug_config.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/stream_info/filter_state.h"
+
+#include "absl/types/optional.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * Configuration for debugging router behavior. The router tries to look up an instance of this
+ * class by its `key`, in FilterState (via StreamInfo). If it's not present, debugging features are
+ * not enabled.
+ *
+ * There is currently no public API for populating this configuration -- neither globally nor
+ * per-request -- users desiring to use the router debugging features should create and install
+ * their own custom StreamDecoderFilter to set DebugConfig as desired before the router consumes
+ * it.
+ *
+ * This is intended to be temporary, and should be replaced by some proper configuration (e.g. in
+ * router.proto) when we get unified matchers. See https://github.com/envoyproxy/envoy/issues/5569.
+ *
+ * TODO(mergeconflict): Keep this promise.
```

Comment:
FWIW, I think this feature can stand independently and you don't need this comment, but up to you :)

Reply:
Yeah, I don't love configuring this feature this way. Maybe there will still be a need for `DebugConfig` even after we have matchers behaving the way we want, but I'll leave the comment until then.